visual-dotnet/SauceLabs.Visual/Models/SelectiveRegion.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using OpenQA.Selenium;
+using SauceLabs.Visual.GraphQL;
+using SauceLabs.Visual.Utils;
+
+namespace SauceLabs.Visual.Models
+{
+    public class SelectiveRegion
+    {
+        public IgnoreRegion? Region { get; set; }
+        public IWebElement? Element { get; set; }
+
+        public DiffingOption[]? EnableOnly { get; set; }
+        public DiffingOption[]? Disable { get; set; }
+
+        internal RegionIn ToRegionIn()
+        {
+            if (Region != null && Element != null)
+            {
+                throw new VisualClientException("ambiguous selective region. Only one of 'region' or 'element' has to be specified.");
+            }
+            if (Region == null && Element == null)
+            {
+                throw new VisualClientException("invalid selective region. one of 'region' or 'element' has to be specified.");
+            }
+
+            var diffingOptions = DiffingOptionsInHelper.CreateFromEnableOnlyDisable(EnableOnly, Disable);
+            return Region != null ? new RegionIn(Region, diffingOptions) : new RegionIn(Element!, diffingOptions);
```

Comment:
Maybe worth using something like this https://github.com/mcintyre321/OneOf. It sounds like that would make sure it is compile time checked.

Reply:
We can merge this now and change it later, as the API is still unstable /  beta.